lib/synapse/haproxy.rb

```diff
@@ -714,6 +714,11 @@ def generate_backend_stanza(watcher, config)
             log.info "synapse: restart required because haproxy_server_options changed for #{backend_name}"
             @restart_required = true
           end
+          if (old_backend.fetch('weight', "") !=
```

Comment:
so, it looks like this section of code is only invoked if the state file is enabled. i think this needs a little re-thinking...